frontend/src/components/main-page/main-page.tsx

```diff
@@ -9,13 +9,15 @@ const MainPage: React.FC = () => {
   const user = useAppSelector((store) => store.profile.user);
   const navigate = useNavigate();
   useEffect(() => {
-    if (!user?.firstName) {
-      navigate(`${MentorMenteeRoute.SETTINGS_PROFILE}/1`);
-      return;
-    }
-    if (!user?.isCompleteTest) {
-      navigate(`${MentorMenteeRoute.SETTINGS_PROFILE}/2`);
-      return;
+    if (user) {
+      if (!user?.firstName) {
```

Comment:
you already checked that user is not null
```suggestion
      if (!user.firstName) {
```